utils/state.go

```diff
@@ -0,0 +1,54 @@
+package utils
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+)
+
+// defaultScuttleFile - Constant default when no override is presented
+// for which file is holding the data at rest. Defaults to `$PWD/.scuttle.json`
+const defaultFile = ".scuttle.json"
+
+// StateManager - An interface to use when stubbing
+// the filesystem operations. Implements the method signatures
+// from ioutil. see structs.go - IOStateManager for a reference
+// implementation
+type StateManager interface {
+	ReadState() (Secrets, error)
+	WriteState(Secrets) error
+}
+
+// IOStateManager - Declare a state manager that uses ioutil to actually invoke file IO.
+// subscribes to the interface declared in state.go line 16, and is not consumed when testing
```

Comment:
I'm generally against adding references to other code via file + line number, since that is able to change in the future and likely would not get updated in the comment